### PR TITLE
add function for querying collections from CMR

### PR DESCRIPTION
**Pull request by tsutterley**
add function for retrieving the maximum available version for a dataset

```
>>> import sliderule.earthdata
>>> sliderule.earthdata.__cmr_max_version('NSIDC_ECS','ATL03')
'006'
```

### CODE REVIEW

File: clients/python/sliderule/earthdata.py [hunk 1]
```diff
@@ -178,6 +178,35 @@ def __cmr_granule_metadata(search_results):
...
+    params = []
+    params.append(f'?provider={provider}')
+    params.append(f'&short_name={short_name}')
+    # full CMR query url
+    cmr_query_url += "".join(params)
+    logger.debug(f'cmr request={cmr_query_url}\n')
+    # ssl context
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
```
Comment (scottyhq) [inline clients/python/sliderule/earthdata.py:198]: out of curiosity what does this do is it necessary?

File: clients/python/sliderule/earthdata.py [hunk 2]
```diff
@@ -178,6 +178,35 @@ def __cmr_granule_metadata(search_results):
...
+    # query CMR for collection metadata
+    req = urllib.request.Request(cmr_query_url)
+    response = urllib.request.urlopen(req, context=ctx)
```
Comment (scottyhq) [inline clients/python/sliderule/earthdata.py:201]: `requests` is a dependency
https://github.com/ICESat2-SlideRule/sliderule/blob/7e16d3e90efc932fc96219a2bba81daf040b44fa/clients/python/requirements.txt#L1

but is urllib easy enough to use instead throughout the client such that requests could be dropped?

**Review by scottyhq — CHANGES_REQUESTED** [review]
Nice! I think this will be helpful. Suggest adding a test to: 

https://github.com/ICESat2-SlideRule/sliderule/blob/main/clients/python/tests/test_earthdata.py

File: clients/python/sliderule/earthdata.py (hunk 1 above)
Comment (tsutterley) [reply clients/python/sliderule/earthdata.py:198]: sometimes urllib throws ssl errors if a context is not provided

File: clients/python/sliderule/earthdata.py (hunk 2 above)
Comment (tsutterley) [reply clients/python/sliderule/earthdata.py:201]: good point. we might want to create an issue to unify how HTTP requests work

**Comment by tsutterley** [comment]
thanks @scottyhq! I added some tests to the `TestCMR` class within `test_earthdata` :+1:

**Comment by scottyhq** [comment]
Looks like some of the tests are failing related to arcticdem file names. But this error as well: `E       AttributeError: module 'sliderule.earthdata' has no attribute '_TestCMR__cmr_max_version'` . might need some guidance from @jpswinski to get the green checkmark!

**Comment by tsutterley** [comment]
Looks like `pytest` doesn't like using functions starting with `__`. Importing the functions without that prefix works with `pytest` on my local machine

File: clients/python/sliderule/earthdata.py (hunk 1 above)
Comment (jpswinski) [reply clients/python/sliderule/earthdata.py:198]: I believe this disables cert verification.  So it protects against man-in-the-middle and things like that.  There are a handful of reasons a valid cert could fail to be verified; and while I know technically we shouldn't do this, it appears in the code for CMR access elsewhere and I don't think it is overly risky as there is no sensitive information being passed in either direction.

**Pull request merged**